### ISSUE TITLE
MixIn `@JsonProperty` for `Object.hashCode()` is ignored

### ISSUE THREAD

**Issue by freddyboucher** [FasterXML/jackson-databind#1990]
Hi,

The following code works well in 2.8.10 but not anymore with 2.8.11, neither with 2.9.x

```java
public class MixinTest {

    @Test
    public void testMixin() throws Exception {
        ObjectMapper objectMapper = new ObjectMapper();
        objectMapper.addMixIn( Object.class, ObjectMixIn.class );

        Assert.assertEquals( "{\"hashCode\":1864350231}", objectMapper.writeValueAsString( new Bean() ) );
    }

    public class Bean {
    }

    @JsonPropertyOrder( alphabetic = true )
    public interface ObjectMixIn {

        @Override
        @JsonProperty
        int hashCode();
    }
}
```

Is it expected? I didn't see anything in the release notes.
Thanks

**Comment by cowtowncoder** [comment]
Ok, first things first: behavior should not have changed in 2.8 patch, and that is unfortunate.

Second: I think I know what kind of change could cause this. Mix-in handling code did allow mix-ins from `java.lang.Object`, with quite a bit of extra handling. When refactoring handling, as part of rewriting annotation introspection system (for unrelated thread-safety issues), it is possible this was dropped. I was speculating that the only possible reason for mix-ins in `java.lang.Object` would be inclusion of `hashCode()` for serialization... looks like I was right on that. :)

So most likely change in 2.9 changed behavior (which itself is not great), but worse, instead of change in minor version, patch was also affected. That is really unfortunate.

**Comment by cowtowncoder** [comment]
Ok, yes. Rewrite of annotation collection for 2.8 did break this somewhat specialized case.
I was able to fix this for 2.9 (and going forward), but since 2.8 is closed (except for critical fixes) I won't try patching it there.

Thank you for reporting this.
